indico/modules/events/registration/controllers/display.py

```diff
@@ -110,20 +111,25 @@ def _process(self):
                        ~RegistrationForm.is_deleted,
                        ~Registration.is_deleted,
                        _join=Registration.registration_form)
-                 .order_by(db.func.lower(Registration.last_name), db.func.lower(Registration.first_name)))
-        registrations = [(reg.get_full_name(), reg.get_personal_data()) for reg in query]
+                 .order_by(*Registration.order_by_name)
+                 .options(contains_eager('registration_form')))
+
+        registrations = [(reg.get_full_name(), reg.get_personal_data(), reg) for reg in query]
         enabled_pd_fields = {field.personal_data_type for reg in regforms for field in reg.active_fields}
         affiliation_enabled = PersonalDataType.affiliation in enabled_pd_fields
         position_enabled = PersonalDataType.position in enabled_pd_fields
+        show_checkin_status = any(reg.registration_form.publish_checkin_enabled and reg.checked_in for reg in query)
         published = bool(RegistrationForm.find(RegistrationForm.publish_registrations_enabled,
                          RegistrationForm.event_id == int(self.event.id)).count())
+
         return self.view_class.render_template(
             'display/participant_list.html',
             self.event,
             event=self.event,
             regforms=regforms,
-            show_affiliation=affiliation_enabled and any(pd.get('affiliation') for reg, pd in registrations),
-            show_position=position_enabled and any(pd.get('position') for reg, pd in registrations),
+            show_affiliation=affiliation_enabled and any(pd.get('affiliation') for fn, pd, reg in registrations),
```

Comment:
thx for spotting that `reg` is actually the name, but `fn` is not a good name (it usually stands for "function"). `name` or just `__` since it's not used anyway might be better